File: [core]/esx_inventory/web/src/lib/nui.ts
/*
 * SPDX-License-Identifier: GPL-3.0-only
 * Copyright (C) 2022-2026 ESX Framework
 */

export function getResourceName(): string {
  const win = window as unknown as { GetParentResourceName?: () => string }
  return win.GetParentResourceName ? win.GetParentResourceName() : "esx_inventory"
}

export async function fetchNui<T = unknown>(event: string, data?: unknown): Promise<T | null> {
  try {
    const response = await fetch(`https://${getResourceName()}/${event}`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(data ?? {}),
    })
    return (await response.json()) as T
  } catch {
    return null
  }
}

File: [core]/esx_multicharacter/client/modules/nui.lua
-- SPDX-License-Identifier: GPL-3.0-only
-- Copyright (C) 2022-2026 ESX Framework

RegisterNuiCallback('SelectCharacter', function (data, cb)
    local selectedIndex = tonumber(data.id)
    
    if selectedIndex then
        Menu:SelectCharacter(selectedIndex)
    end
    cb('ok')
end)

RegisterNuiCallback('PlayCharacter', function (data, cb)
    Menu:PlayCharacter()
    cb('ok')
end)

RegisterNuiCallback('DeleteCharacter', function (data, cb)
    Menu:DeleteCharacter()
    cb('ok')
end)

RegisterNuiCallback('CreateCharacter', function (data, cb)
    Menu:NewCharacter()
    cb('ok')
end)